### PR TITLE
vscode-extensions: Update/add extensions

### PR DESCRIPTION
**Pull request by kubukoz**
Updates the following extensions:

- ms-azuretools.vscode-docker: 0.8.1 -> 1.9.1
- scala-lang.scala: 0.4.5 -> 0.5.1
- scalameta.metals: 1.9.10 -> 1.9.13

Adds the following extensions:

- alygin.vscode-tlaplus: 1.5.3
- baccata.scaladex-search: 0.0.1
- codezombiech.gitignore: 0.6.0
- edonet.vscode-command-runner: 0.0.116
- github.vscode-pull-request-github: 0.22.0
- mishkinf.goto-next-previous-member: 0.0.5
- shyykoserhiy.vscode-spotify: 3.2.1
- slevesque.vscode-multiclip: 0.1.5
- timonwong.shellcheck: 0.12.3
- xyz.local-history: 1.8.1
- yzhang.markdown-all-in-one: 3.4.0
- zhuangtongfa.material-theme: 3.9.12

Remaining changes are a result of automatic reformatting via the nixfmt
extension.

###### Motivation for this change

Wanting newer / more extensions without maintaining a list of all the versions/shas myself ;)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] ~Tested execution of all binary files (usually in `./result/bin/`)~ - doesn't apply
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vscode-extensions/default.nix [hunk 1]
```diff
@@ -714,40 +745,69 @@ let
...
       };
 
       xaver.clang-format = buildVscodeMarketplaceExtension {
         mktplcRef = {
           name = "clang-format";
           publisher = "xaver";
           version = "1.9.0";
-          sha256 = "abd0ef9176eff864f278c548c944032b8f4d8ec97d9ac6e7383d60c92e258c2f";
+          sha256 =
+            "abd0ef9176eff864f278c548c944032b8f4d8ec97d9ac6e7383d60c92e258c2f";
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/default.nix:757]:
```suggestion
          sha256 = "abd0ef9176eff864f278c548c944032b8f4d8ec97d9ac6e7383d60c92e258c2f";
```

nix-fmt is great but 80 line length is not great.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
To comply with the contributing guide each package update and addition needs to be its own commit. The formatting changes should also get their own commit.

**Comment by kubukoz** [comment]
> To comply with the contributing guide each package update and addition needs to be its own commit. The formatting changes should also get their own commit.

Understandable, I'll split it tomorrow :) and I guess I'll get rid of the formatting changes too.

**Comment by kubukoz** [comment]
Took some time but I think it matches the requirements now :) @SuperSandro2000

**Comment by kubukoz** [comment]
Can we get this merged? I have to fix conflicts every now and then and it looks like all the comments have been addressed.

**Comment by kubukoz** [comment]
Thanks!

**Pull request merged**